cl/corpus_importer/management/commands/columbia_merge.py

```diff
@@ -0,0 +1,865 @@
+    Docket.COLUMBIA_AND_RECAP_AND_IDB_AND_HARVARD,
+    Docket.COLUMBIA_AND_SCRAPER_AND_IDB_AND_HARVARD,
+    Docket.COLUMBIA_AND_RECAP_AND_SCRAPER_AND_IDB_AND_HARVARD,
+]
+
+
+VALID_MERGED_SOURCES = [
+    key
+    for key in dict(SOURCES.NAMES).keys()
+    if SOURCES.COLUMBIA_ARCHIVE in key
+]
+
+
+def clean_opinion_content(content: str, is_harvard: bool) -> str:
+    """Strip all non-alphanumeric characters
+    :param content: content from opinion
+    :param is_harvard: true if content is from harvard
+    :return: cleaned content
+    """
+    soup = BeautifulSoup(content, features="html.parser")
+
+    if is_harvard:
+        for op in soup.select("opinion"):
+            # Remove any author tag inside opinion
+            for extra in op.find_all(["author"]):
+                extra.extract()
+
+    prep_text = re.sub(
+        r"[^a-zA-Z0-9 ]", "", soup.getText(separator=" ").lower()
+    )
+    prep_text = re.sub(" +", " ", prep_text)
+    return prep_text
+
+
+def get_cl_opinion_content(cluster_id: int) -> list[dict[Any, Any]]:
+    """Get the opinions content for a cluster object
+    :param cluster_id: Cluster ID for a set of opinions
+    :return: list with opinion content from cl
+    """
+    cl_cleaned_opinions = []
+    opinions_from_cluster = Opinion.objects.filter(cluster_id=cluster_id)
+    is_harvard = False
+
+    for i, op in enumerate(opinions_from_cluster):
+        content = None
+        if len(op.xml_harvard) > 1:
+            content = op.xml_harvard
+            is_harvard = True
+        elif len(op.html_with_citations) > 1:
+            content = op.html_with_citations
+        elif len(op.html_columbia) > 1:
+            content = op.html_columbia
+        elif len(op.html_lawbox) > 1:
+            content = op.html_lawbox
+        elif len(op.plain_text) > 1:
+            content = op.plain_text
+        elif len(op.html) > 1:
+            content = op.html
+
+        if not content:
+            raise EmptyOpinionException("No content in opinion")
+
+        prep_text = clean_opinion_content(content, is_harvard=is_harvard)
+        cl_cleaned_opinions.append(
+            {
+                "id": op.id,
+                "byline": op.author_str,
+                "type": op.type,
+                "opinion": prep_text,
+            }
+        )
+
+    return cl_cleaned_opinions
+
+
+def update_matching_opinions(
+    matches: dict, cl_cleaned_opinions: list, columbia_opinions: list
+) -> None:
+    """Store matching opinion content in html_columbia
+    :param matches: dict with matching position from cl and columbia opinions
+    :param cl_cleaned_opinions: list of cl opinions
+    :param columbia_opinions: list of columbia opinions
+    """
+    for columbia_pos, cl_pos in matches.items():
+        file_opinion = columbia_opinions[columbia_pos]  # type: dict
+        file_byline = file_opinion.get("byline")
+        cl_opinion = cl_cleaned_opinions[cl_pos]
+        opinion_id_to_update = cl_opinion.get("id")
+
```

Comment:
what is this doing? is this changing the Columbia data itself?

Reply:
no, it only saves the columbia opinion in html_columbia field from opinion object as extracted from the file and converted to html